### PR TITLE
Add advanced config to support key overloading

### PR DESCRIPTION
**Pull request by rmtuckerphx**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by rmtuckerphx** [comment]
# Simple Scenario
For a basic user persistence scenario in a skill, simply setting the table name will create a table to is made to store ony user data items.

`config.js`
```
{
    DynamoDb: {
      tableName: `myapp-table-users`,
    }
}
```
The resulting table would be:

| Partition key (userId) | Sort key | userData (Map) | 
|------------------------|----------|----------------|
| amzn1.ask.account.AAAA | -        | JSON           |
| amzn1.ask.account.BBBB | -        | JSON           |


# Key Overloading Scenario
One of the strengths of DynamoDB is the ability to overload keys and store items of different shapes in the same table. We may want to define a sortKey and prefix the primaryKey so that we can store other data. We might want to take advantage of the primaryKey with other sortKey values while still having easy access to the primary user data inside the Jovo-based voice app.
See [Amazon DynamoDB deep dive: Advanced design patterns](https://www.youtube.com/watch?v=6yqfmXiZTlM)

`config.js`
```
{
    DynamoDb: {
      tableName: `myapp-table-adv`,
      primaryKeyColumn: 'PK',
      primaryKeyPrefix: 'USER#',
      sortKeyColumn: 'SK',
      sortKey: 'USER#jovo.user.data',
    }
}
```

The resulting table would be:

| Partition key (PK) | Sort key (SK) | userData (Map) | 
|------------------------|----------|----------------|
| USER#amzn1.ask.account.AAAA | USER#jovo.user.data        | JSON           |
| USER#amzn1.ask.account.BBBB | USER#jovo.user.data        | JSON           |

The Jovo User Data persistence would work the same that it always does, but with this new structure. Any additional data added or accessed on the table would be the responsibility of the developer and would be outside the currently-provided functionality.

Here is a sample table that shows Jovo User Data items mixed with other items int he same table:

| Partition key (PK)          | Sort key (SK)       | userData (Map) | otherAttrib1 | otherAttrib2 |
|-----------------------------|---------------------|----------------|--------------|--------------|
| USER#amzn1.ask.account.AAAA | USER#jovo.user.data | JSON           |              |              |
| USER#amzn1.ask.account.BBBB | USER#jovo.user.data | JSON           |              |              |
| USER#amzn1.ask.account.BBBB | OTHER#key           | -              | value        |              |
| KEY1#key                    | -                   | -              | -            | value        |

File: jovo-integrations/jovo-db-dynamodb/src/DynamoDb.ts [hunk 1]
```diff
@@ -9,6 +9,9 @@ export interface Config extends PluginConfig {
   tableName?: string;
   createTableOnInit?: boolean;
   primaryKeyColumn?: string;
+  primaryKeyPrefix?: string;
```
Comment (aswetlow) [inline jovo-integrations/jovo-db-dynamodb/src/DynamoDb.ts:12]: Please sort the properties alphabetically.
https://travis-ci.org/github/jovotech/jovo-framework/jobs/688064719

File: jovo-integrations/jovo-db-dynamodb/src/DynamoDb.ts [hunk 2]
```diff
@@ -127,11 +133,16 @@ export class DynamoDb implements Db {
     const getDataMapParams: DocumentClient.GetItemInput = {
       ConsistentRead: true,
       Key: {
-        [this.config.primaryKeyColumn!]: primaryKey,
+        [this.config.primaryKeyColumn!]: `${this.config.primaryKeyPrefix!}${primaryKey}`,
```
Comment (aswetlow) [inline jovo-integrations/jovo-db-dynamodb/src/DynamoDb.ts:136]: The default scenario (without the prefix) would result in `undefined${primaryKey}` here. Is that correct? I would use '' as default value for `this.config.primaryKeyPrefix`.

**Comment by aswetlow** [comment]
Hey Mark! Thank you for the PR and the great explanation. I left some comments.

**Comment by rmtuckerphx** [comment]
I made formatPrimaryKey so that I could easily use the same formatting to create the primary key so it could be used in adding other rows in the advanced scenario.

How would I access this in a handler?
`this.$db.formatPrimaryKey('myKey');`

**Pull request merged**